bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/protocol/CommonCommands.java

```diff
@@ -0,0 +1,460 @@
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) 0x00);// PUTIN
+
+        macacoFrame.add((byte) (idNode));// Start Offset
+        macacoFrame.add((byte) ((byte) slot + 5)); // Number Of byte of payload= command + set byte
+
+        for (var i = 0; i <= slot - 1; i++) {
+            macacoFrame.add((byte) 00); // pongo a zero i byte precedenti lo
+                                        // slot da modificare
+        }
+        macacoFrame.add(shortCommand);// PAYLOAD
+
+        macacoFrame.add((byte) 0x0);// Empty - Temperature Measured Value
+        macacoFrame.add((byte) 0x0);// Empty - Temperature Measured Value
+        macacoFrame.add(byte1);// Temperature Setpoint Value
+        macacoFrame.add(byte2);// Temperature Setpoint Value
+
+        logger.debug("sendFORCEFrame - {}, soulissNodeIPAddressOnLAN: {}", macacoToString(macacoFrame), gwConfig);
+        queueToDispatcher(macacoFrame, gwConfig);
+    }
+
+    public final void sendDBStructFrame(GatewayConfig gwConfig) {
+        ArrayList<Byte> macacoFrame = new ArrayList<>();
+        macacoFrame.add((byte) SoulissUDPConstants.SOULISS_UDP_FUNCTION_DBSTRUCT_REQ);
+        macacoFrame.add((byte) 0x0);// PUTIN
+        macacoFrame.add((byte) 0x0);// PUTIN
+        macacoFrame.add((byte) 0x0);// Start Offset
+        macacoFrame.add((byte) 0x0); // Number Of
+
+        logger.debug("sendDBStructFrame - {}, soulissNodeIPAddressOnLAN: {}", macacoToString(macacoFrame), gwConfig);
+        queueToDispatcher(macacoFrame, gwConfig);
+
+        // Note:
+        // Structure of DBStructFrame:
+        // nodes = mac.get(5);
+        // maxnodes = mac.get(6);
+        // maxTypicalXnode = mac.get(7);
+        // maxrequests = mac.get(8);
+        // MaCacoIN_s = mac.get(9);
+        // MaCacoTyp_s = mac.get(10);
+        // MaCacoOUT_s = mac.get(11);
+    }
+
+    /*
+     * Queue command to Dispatcher (for securesend retransmission)
+     */
+    private final void queueToDispatcher(ArrayList<Byte> macacoFrame, GatewayConfig gwConfig) {
+        ArrayList<Byte> buf = buildVNetFrame(macacoFrame, gwConfig.gatewayLanAddress, (byte) gwConfig.userIndex,
+                (byte) gwConfig.nodeIndex);
+        byte[] merd = toByteArray(buf);
+
+        InetAddress serverAddr;
+        try {
+            serverAddr = gwConfig.gatewayWanAddress.isEmpty() ? InetAddress.getByName(gwConfig.gatewayLanAddress)
+                    : InetAddress.getByName(gwConfig.gatewayWanAddress);
+            var packet = new DatagramPacket(merd, merd.length, serverAddr,
+                    SoulissUDPConstants.SOULISS_GATEWAY_DEFAULT_PORT);
+            SendDispatcherRunnable.put(packet, logger);
+        } catch (IOException e) {
+            logger.warn("Error: {} ", e.getMessage());
+        }
+    }
+
+    /*
+     * send broadcast UDP frame - unused in this version - to Test
+     */
+    private final void sendBroadcastNow(ArrayList<Byte> macacoFrame) {
```

Comment:
Can you elaborate what's the purpose of this test method and when you'd use it?

Reply:
removed to Test comment. It's unused . stop.